### DISCUSSION TITLE
RFC: Plugin change log

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#8460] [Ideas]
Could there be a standard plugin changelog mechanism? I find it not confident to upgrade plugins because I don't know what is changed.

If put `$:/plugins/linonetwo/mobile-imessage-layout/change-log` in a plugin, then show it when upgrade it. (Not when fresh install it).

And it always only include changes from last update. Not including full change from v0.0.1. (Old changes can be find in the git history so it is OK to delete them).

**Comment by twMat** [comment]
One idea would be a button to create overwrites for every shadow tiddler in the pre-upgrade version. Then after upgrade, you can compare. Not a great solution but should do the job. Maybe you could tweak the import mechanism to do it automatically.

You might also benefit from my [Overwrite plugin](https://overwrite.tiddlyhost.com/)

**Comment by Jermolene** [comment]
Hi @linonetwo yes, I like the idea – one of the motivations of introducing the plugin stability badges is that I plan to move the experimental plugins out of the main release process so that they can have ad hoc releases. I have written up the plans in #8477
